src/material-experimental/mdc-slider/slider.ts

```diff
@@ -157,11 +157,13 @@ export class MatSlider implements AfterViewInit, OnDestroy {
     }
 
   ngAfterViewInit() {
-    this._foundation.init();
-    if (this._platform.isBrowser) {
-      this._foundation.layout();
-    }
-    this._initialized = true;
+    this._initInputs().then(() => {
```

Comment:
Can we avoid the `Promise` here? These sorts of async actions make it more difficult to test and we can't easily remove it in the future once people start depending on it. Maybe the "changed after checked" error can be worked around by setting a value directly in the DOM, rather than going through Angular.

Reply:
The value does get set directly in the DOM
https://github.com/angular/components/blob/29f119ee4e49d29720578cf9981b8531bf86dbf8/src/material-experimental/mdc-slider/slider-thumb.ts#L60

Reply:
Then there shouldn't be a risk of it triggering a "changed after checked" error, unless it was being caused by something else.

Reply:
I'd be up to VC today and try and remove the `Promise` here, though